### ISSUE TITLE
Type becomes `@UnknownInitialization` after switch

### ISSUE THREAD

**Issue by cushon** [typetools/checker-framework#6260]
The following program is accepted by the nullness checker in 3.38.0, and rejected in 3.39.0.

Is this a regression or a deliberate change?

```java
enum E {
  FOO;

  E getIt() {
    return FOO;
  }

  String go() {
    E e = getIt();
    switch (e) {
      case FOO:
        return "foo";
    }
    throw new AssertionError(e);
  }
}
```

```
$ ./checker-framework-3.38.0/checker/bin/javac -processor Nullness E.java
```

```
$ ./checker-framework-3.39.0/checker/bin/javac -processor Nullness E.javaE.java:14: error: [argument] incompatible argument for parameter detailMessage of AssertionError constructor.
    throw new AssertionError(e);
                             ^
  found   : @UnknownInitialization @Nullable E
  required: @Initialized @Nullable Object
1 error
```

**Comment by smillst** [comment]
I improved the handling of switches in 3.39.0, so now dataflow treats `throw new AssertionError(e);` as dead code and does not analysis it.  So the type of `e` in `new AssertionError(e)` is not refined and the checker issues a false positive error.   I've fixed this in https://github.com/typetools/checker-framework/pull/6246 which will be in the next release.
